frontend/src/lib/components/EntityFilterInfo.tsx

```diff
@@ -0,0 +1,41 @@
+import { ActionFilter, EntityFilter, EntityTypes, FunnelStepRangeEntityFilter } from '~/types'
+import { Typography } from 'antd'
+import React from 'react'
+import { TextProps } from 'antd/es/typography/Text'
+import { getKeyMapping } from 'lib/components/PropertyKeyInfo'
+import { getDisplayNameFromEntityFilter } from 'scenes/insights/utils'
+
+interface Props {
+    filter: EntityFilter | ActionFilter | FunnelStepRangeEntityFilter
+}
+
+function TextWrapper(props: TextProps): JSX.Element {
+    return (
+        <Typography.Text ellipsis={true} style={{ maxWidth: 400 }} {...props}>
+            {props.children}
+        </Typography.Text>
+    )
+}
+
+export function EntityFilterInfo({ filter }: Props): JSX.Element {
+    const title = getDisplayNameFromEntityFilter(filter)
+    const subtitle = getDisplayNameFromEntityFilter(filter, false)
+
+    if (filter.type === EntityTypes.NEW_ENTITY || (!title && !subtitle)) {
+        return <TextWrapper title="Select filter">Select filter</TextWrapper>
+    }
+
+    const titleTranslation = getKeyMapping(title, 'event')?.label ?? title ?? undefined
+    const subTitleTranslation = getKeyMapping(subtitle, 'event')?.label ?? subtitle ?? undefined
```

Comment:
Translation feels like a misnomer here (particularly with strings). Maybe `titleToDisplay`?

Reply:
sounds much better. english is hard